### ISSUE TITLE
Poetry complains about sqlalchemy-utils restriction ~=3.6

### ISSUE THREAD

**Issue by staticdev** [kvesteri/sqlalchemy-utils#663]
If you just change it to >=3.6 will fix.

**Comment by kurtmckee** [comment]
Please create a repository that recreates this issue. I would like to see how this is happening.

**Comment by staticdev** [comment]
@kurtmckee done https://github.com/staticdev/sandbox/blob/main/pyproject.toml
just enter the folder and try to resolve dependencies with command `poetry lock`, you will get this output:

```
Updating dependencies
Resolving dependencies... (0.1s)

The current project's Python requirement (>=3.11) is not compatible with some of the required packages Python requirement:
  - sqlalchemy-utils requires Python ~=3.6, so it will not be satisfied for Python >=4.0

Because no versions of sqlalchemy-utils match >0.38.3
 and sqlalchemy-utils (0.38.3) requires Python ~=3.6, sqlalchemy-utils is forbidden.
So, because sandbox depends on SQLAlchemy-Utils (>=0.38.3), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties
    
    For sqlalchemy-utils, a possible solution would be to set the `python` property to ">=3.11,<4.0"
```

**Comment by staticdev** [comment]
I also created the PR with the fix https://github.com/kvesteri/sqlalchemy-utils/pull/664

**Comment by kurtmckee** [comment]
I saw the PR, thanks for that! I'll take a look at this as quick as I can.

**Comment by kurtmckee** [comment]
Although I've released a new version that should include support for Python 3.11, I want to understand this problem better. I haven't worked on a project that uses setuptools in a long time, and I'm still surprised that the documented behavior for `~=3.6` isn't producing the expected results.

I'm going to leave this open because it still needs to be addressed, possibly with `>=3.6,<4` (which, again, is what `~=3.6` is the documented equivalent of).

**Comment by kurtmckee** [comment]
Ah, I see it now. Thanks for posting that example `pyproject.toml` file! It let me recreate the problem.

This is a problem with the Python requirement specified in the `pyproject.toml` file, not a problem with sqlalchemy-utils.

sqlalchemy-utils declares support for Python 3 versions >= 3.6, and this is communicated with the `~=3.6` syntax. Poetry is correctly interpreting this as `>=3.6,<4.0`. (I was worried there was a problem in this regard.)

However, the Python support declared in `pyproject.toml` is anything greater than Python 3.11, including Python 4 versions. sqlalchemy-utils doesn't declare support for Python 4, so Poetry's dependency resolver is rejecting this combination.

Please modify your projects as indicated in the Poetry error message.

```toml
[tool.poetry.dependencies]
python = ">=3.11,<4.0"
```

This will resolve the problem.
